src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -769,6 +774,19 @@ private function registerRequestConfiguration(array $config, ContainerBuilder $c
         }
     }
 
+    private function registerEncryptedSecretsConfiguration(array $config, ContainerBuilder $container)
+    {
+        $secretsFile = $config['encrypted_secrets']['secrets_file'];
+        $publicKeyFile = $config['encrypted_secrets']['public_key_file'];
+        $privateKeyFile = $config['encrypted_secrets']['private_key_file'];
+        $secretsHandler = new JweHandler($secretsFile, $publicKeyFile, $privateKeyFile);
+        //TODO: this will only validate presence/size of files, make sure the configuration can decrypt values before runtime
+
+        $container->fileExists($secretsFile);
+        $container->getDefinition('secret_var_processor')
+            ->addMethodCall('enableSecretsLookup', [$secretsFile, $publicKeyFile, $privateKeyFile]);
```

Comment:
We should just use proper DI here: make the `JweHandler` a constructor arg to the SecretsVarProcessor and set it normally.

Reply:
yeah all of this logic was to validate that the `enableSecretsLookup` call would not blow up at runtime, if we can skip that then I can remove all of this and configure through xml